### ISSUE TITLE
[Minor feature request] Manifest for download dependencies

### ISSUE THREAD

**Issue by CaelumLaron** [TheLartians/CPM.cmake#160]
Hello, first thanks for your work, it is very good and helps me a lot in my projects. 

I think that including a manifest as a way to download dependencies would be an incredible resource for this project, as it will improve the organization and readability of the projects that use it as a package manager.

**Comment by Andres6936** [comment]
You are right, in my opinion CPM marks a new starting point for every C/C++ project, even if this involves the adherence to guidelines, I remember that for Issue #148 , I intended to extend in the explanation, but the lack of time limited me, and a manifesto would be helpful not only for this project, but for many others.

In fact, I am in the task of defining basic aspects that must be fulfilled in order to take advantage of the use of a package manager such as CPM.

Here is a sample of what he planned to do.

![WhatIsProject](https://user-images.githubusercontent.com/18540634/96810370-2c7eaa00-13e1-11eb-9f00-584bde663b4e.png)

**Comment by TheLartians** [comment]
Hey thanks for the suggestion! To be clear, with a manifest, do you mean something like a JSON that lists the first-level dependencies of the project, say similar to npm or vcpkg packages?

**Comment by CaelumLaron** [comment]
Yes @TheLartians! I don't know if there is any road map or this project, but including it definitely can help a lot.

**Comment by TheLartians** [comment]
I do see the value in having a cleaner project structure like that, though the issue I see is that this declarative approach is quite contrary to the imperative way that CMake works. So to implement this "properly" would require the package manager to work on a higher level and invoke CMake itself, or we can restrict the manifest to the dependencies only. Though in the latter case you could already just create a `dependencies.cmake` file which is imported from the main `CMakeLists.txt`, that only adds the dependencies. I'll have to think about this a bit.
